### PR TITLE
General fixes

### PR DESCRIPTION
**Pull request by jgradim**

### CODE REVIEW

File: app/components/forms/RecoverPasswordForm.jsx [hunk 1]
```diff
@@ -1,64 +1,66 @@
...
     return (
-      <form className="pure-form" onSubmit={this.onSubmit.bind(this)}>
-        <FormError error={this.state.errors.serverError} />
+      <form className="pure-form pure-form-narrow" onSubmit={this.onSubmit}>
+        <FormError error={errors.serverError} />
+
         <fieldset>
           <input
-            onChange={this.changeEventHandler("email")}
+            id="email"
```
Comment (jgradim) [inline app/components/forms/RecoverPasswordForm.jsx]: remove

File: app/components/pages/Confirmation.jsx [hunk 2]
```diff
@@ -1,65 +1,57 @@
...
-        this.setState({ errors: errorTransform(error, { 404: "confirmation token not found" }) });
-      });
-  }
+  confirm = () => {
+    const { confirmAccount, match: { params } } = this.props;
 
+    confirmAccount(params.id, params.token)
+    .then(() => this.setState({ confirmed: true }))
+    .catch(error => {
+      this.setState({ errors: errorTransform(error, { 404: "confirmation token not found" }) });
```
Comment (jgradim) [inline app/components/pages/Confirmation.jsx]: missing translation

File: app/components/pages/RecoverPassword.jsx [hunk 3]
```diff
@@ -1,85 +1,57 @@
...
-    let content = !this.state.sent ? this.renderForm() : this.renderSentMessage();
+    const { t } = this.props;
+    const { sent, email } = this.state;
 
     return (
       <div className="posts">
         <section className="post">
           <header className="post-header">
-            <h1 className="post-title">Programa de Mentorados de Engenharia Informática - FEUP</h1>
+            <h1 className="post-title">{t("recover_password:title")}</h1>
```
Comment (jgradim) [inline app/components/pages/RecoverPassword.jsx:31]: check if working, wrong namespace

File: app/components/pages/RecoverPassword.jsx [hunk 4]
```diff
@@ -1,85 +1,57 @@
...
         <section>
-          <h2>Contacto</h2>
+          <h2>{t("home:contact.title")}</h2>
 
-          <p>
-            Se quiserdes contactar as pessoas responsaveis pela manutencao da
-            plataforma, com queixas, sugestoes, ou qualquer cena, mandai email
-            para <a href={`mailto:${email}`}>{email}</a>.
-          </p>
+          <p dangerouslySetInnerHTML={{ __html: t("home:contact.content", { email: contactEmail }) }} />
```
Comment (jgradim) [inline app/components/pages/RecoverPassword.jsx:47]: extract contact email to a constant

File: app/components/pages/ResetPassword.jsx [hunk 5]
```diff
@@ -29,75 +31,77 @@ class ResetPassword extends React.Component {
...
     return (
-      <div className="page-registration">
-        <Section title="Reset password">
-          <div className="pure-g">
-            <div className="pure-u-1-5">
+      <div className="posts">
+        <header className="post-header">
+          <h1 className="post-title">
+            {recovered || !valid ? t("already_reset.title") : t("not_reset.title")}
+          </h1>
```
Comment (jgradim) [inline app/components/pages/ResetPassword.jsx]: Close `<header>` here

File: app/components/pages/ResetPassword.jsx [hunk 6]
```diff
@@ -29,75 +31,77 @@ class ResetPassword extends React.Component {
...
   }
 
-  onSuccessfulReset(data) {
-    this.context.session.doLogin({ jwt: this.props.match.params.token });
+  onSuccessfulReset = (data) => {
+    const { setJWT, getCurrentUser, match, t } = this.props;
+
+    setJWT(match.params.token);
+    addToast({ content: t("toasts:reset_pw_success"), level: TOAST_LEVEL_SUCCESS });
+    getCurrentUser();
```
Comment (jgradim) [inline app/components/pages/ResetPassword.jsx]: only keep `addToast`

File: app/components/pages/ResetPassword.jsx [hunk 7]
```diff
@@ -29,75 +31,77 @@ class ResetPassword extends React.Component {
...
 }
 
-export default ResetPassword;
+export default compose(
+  translate([ "reset_password", "toasts" ]),
+
+  connect(
+    () => ({}),
+    {
+      setJWT,
```
Comment (jgradim) [inline app/components/pages/ResetPassword.jsx]: remove

File: app/components/pages/ResetPassword.jsx [hunk 8]
```diff
@@ -29,75 +31,77 @@ class ResetPassword extends React.Component {
...
+      getCurrentUser,
```
Comment (jgradim) [inline app/components/pages/ResetPassword.jsx]: remove

File: app/lib/api.js [hunk 9]
```diff
@@ -35,6 +35,7 @@ export const users = {
     return api.patch(`/users/${id}`, { user: attributes }, { headers });
   },
   me: () => api.get("/users/me"),
+  confirm: (id, confirmation_token) => api.post(`/users/${id}/confirm`, { confirmation_token }),
```
Comment (jgradim) [inline app/lib/api.js]: remove

File: app/translations/en.yml [hunk 10]
```diff
@@ -103,9 +98,39 @@ account:
...
+  not_reset:
+    title: Reset password
+  recovered:
+    email1: We've sent an email to {{email}}. Click the link in the email to reset your password.
+    email2: If you don't see the email, check other places it might be, like your junk, spam, social, or other folders.
+  form:
+    password:
+      placeholder: New password
+    submit: Change password
+    error: Login not found
```
Comment (jgradim) [inline app/translations/en.yml]: Account not found

File: app/translations/pt.yml [hunk 11]
```diff
@@ -103,9 +99,39 @@ account:
...
+  not_reset:
+    title: Recuperar password
+  recovered:
+    email1: Foi enviado um email para {{email}}. Carrega no link no email para recuperar a tua password.
+    email2: Se não vires o email, verifica outros lugares onde possa estar, como o lixo, spam, social, ou outras pastas.
+  form:
+    password:
+      placeholder: Nova password
+    submit: Mudar password
+    error: Login não encontrado
```
Comment (jgradim) [inline app/translations/pt.yml]: Conta não encontrada

File: app/components/pages/ResetPassword.jsx [hunk 12]
```diff
@@ -1,22 +1,24 @@
...
+import { translate } from "react-i18next";
+import { connect } from "react-redux";
 
 import { password_recovery_tokens } from "lib/api";
 import Section from "components/elements/Section";
 import ResetPasswordForm from "components/forms/ResetPasswordForm";
 import RecoverPasswordForm from "components/forms/RecoverPasswordForm";
 
+import { setJWT, getCurrentUser } from "actions/session";
+import { addToast, TOAST_LEVEL_SUCCESS } from "actions/toasts";
```
Comment (jgradim) [inline app/components/pages/ResetPassword.jsx]: Add curried `addSuccessToast`, `addFailureToast`

File: app/components/Layout.jsx [hunk 13]
```diff
@@ -35,9 +35,10 @@ class Layout extends React.Component {
   getUserAndMeta() {
     const { getMeta, getCurrentUser } = this.props;
 
+    this.setState({ loading: true });
     Promise.all([
       ignoreRejection(getCurrentUser()),
-      getMeta(),
+      ignoreRejection(getMeta()),
```
Comment (jgradim) [inline app/components/Layout.jsx]: rename to `ignoreFailure`

File: app/components/pages/RecoverPassword.jsx (hunk 3 above)
Comment (jgradim) [reply app/components/pages/RecoverPassword.jsx:31]: working, added namespace explicitly nonetheless

**Comment by jgradim** [comment]
@hugopeixoto updated!

File: app/actions/session.js [hunk 14]
```diff
@@ -52,5 +54,19 @@ export const logout = () => {
...
     dispatch(clearJWT());
     dispatch(clearCurrentUser());
+    dispatch(clearMeta());
+  };
+};
+
+export const resetPassword = (id, password, token) => {
+  return dispatch => {
+    return users
+      .update(id, { password }, token )
```
Comment (hugopeixoto) [inline app/actions/session.js]: There's extra whitespace before the `)`

**Pull request merged**